### PR TITLE
feat(mi): Participants create/edit analytics (M2-6735)

### PR DESCRIPTION
**Pull request by farmerpaul**
- [x] Tests for the changes have been added
- [x] Delivered the `fix` or `feature` branches into `develop` or `release` branches via `Squash and Merge` (to keep clean history)

### 📝 Description

🔗 [Jira Ticket M2-6735](https://mindlogger.atlassian.net/browse/M2-6735)

This PR tracks new/updated Mixpanel events related to adding and editing participants.

Also reduces browser console warnings due to an unrelated misspelt `data-testid` prop.

### 🪤 Peer Testing

Please refer to [JIRA ticket](https://mindlogger.atlassian.net/browse/M2-6735) for details around when events are triggered and what event properties should be attached to those events.

#### Prerequisites:

- To test, you'll need to have access to the **MindLogger Analytics Dev** project in Mixpanel; please let @farmerpaul know if you are planning to do peer testing so that you can be granted access.
- Ensure you have these env vars set locally:
  ```
  REACT_APP_ENV=dev
  REACT_APP_MIXPANEL_TOKEN=c1e1e662f03d9f1363768731077ea000
  REACT_APP_MIXPANEL_FORCE_ENABLE=true
  REACT_APP_DEVELOP_BUILD_VERSION=head
  ```

#### Testing:
* For each table row in the [JIRA ticket](https://mindlogger.atlassian.net/browse/M2-6735):
  1. Follow the instructions under the **Trigger** column to trigger the event.
  2. Open the **MindLogger Analytics Dev** project in Mixpanel, and navigate to the [Events tab](https://mixpanel.com/project/3333580/view/3841316/app/events). Refresh if needed.

      **Expected outcome:** The event matching the **New Event Name** column should have been added to the top of the list.

  4. Expand the event and open the **Your Properties** tab:  
      Screenshot 2024-06-18 at 4 32 53 PM

      **Expected outcome:** The custom properties for that event mentioned in the **Details** column should be listed, as well as **Applet ID** as the JIRA ticket says applies to all events.
* Click the link under the **Distinct ID** column for any of these recently logged events.

    **Expected outcome:** Under the User Profile Properties box on the left, there should be a field called **App Build Number** whose value is **head**.

### CODE REVIEW

**Review by sultanofcardio — COMMENTED** [review]
I've had a look through the code. Everything looks alright, just waiting to test it now

File: src/shared/utils/mixpanel/mixpanel.types.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export enum MixProperties {
```
Comment (sultanofcardio) [inline src/shared/utils/mixpanel/mixpanel.types.ts]:
```suggestion
export enum MixPanelProperties {
```

File: src/shared/utils/mixpanel/mixpanel.types.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  MultiInformantAssessmentId = 'Multi-informant Assessment ID',
+  SourceAccountType = 'Source Account Type',
+  TargetAccountType = 'Target Account Type',
+  InputAccountType = 'Input Account Type',
+  IsSelfReporting = 'Is Self Reporting',
+  Tag = 'Tag',
+  Via = 'Via',
+}
+
+export type MixPayload = Partial<Record<MixProperties, unknown>>;
```
Comment (sultanofcardio) [inline src/shared/utils/mixpanel/mixpanel.types.ts]:
```suggestion
export type MixPanelPayload = Partial<Record<MixProperties, unknown>>;
```

**Review by sultanofcardio — APPROVED** [review]
Looking good 🚀

File: src/shared/utils/mixpanel/mixpanel.types.ts (hunk 1 above)
Comment (farmerpaul) [reply src/shared/utils/mixpanel/mixpanel.types.ts]: Agreed. I picked this name in an attempt to have alignment with the web app which already introduces a type with this name, but I think I'll opt to change it in the web app as well. 👍🏻

**Pull request merged**